alacritty/src/input/mod.rs

```diff
@@ -1050,12 +1050,9 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
         let point = self.ctx.mouse().point(&self.ctx.size_info(), display_offset);
         let hyperlink = self.ctx.terminal().grid()[point].hyperlink();
 
-        // Function to check if mouse is on top of a hint.
-        let hint_highlighted = |hint: &HintMatch| hint.should_highlight(point, hyperlink.as_ref());
-
         if let Some(mouse_state) = self.message_bar_cursor_state() {
             mouse_state
-        } else if self.ctx.display().highlighted_hint.as_ref().map_or(false, hint_highlighted) {
+        } else if hyperlink.is_some() {
```

Comment:
This change isn't strictly necessary, but it seems to just work so why not?

Reply:
Hm, the reason this works is because the logic is bit inconsistent. We basically set the cursor from the renderer and from this function, so that's why the code here works even though it doesn't check the regular region anyway.

In the current state of things you can remove all the logic around `hyperlink` and `::Pointer` and it'll still work. 

I don't really like that it's like that though since it's a bit error prone?

Reply:
Pretty much yes. I'm not sure what to do about this function though, because clearly the check could be removed, though, maybe there's an edge case...

Reply:
Yeah honestly I was kinda worried that this patch would only change the mouse cursor when on a hyperlink escape, but that doesn't seem to be the case.

Reply:
Actually I found why this is necessary and my changes broke things:

If you're in mouse mode and hover over a URL, this will change mouse shape without shift.